### DISCUSSION TITLE
What is the best commercial solver for nonlinear equality constraints?

### DISCUSSION THREAD

**Discussion by 0xF0F0F0F** [yalmip/YALMIP#1317] [General discussion] [answered]
What is the best nonconvex commercial solver for nonlinear equality constraints? such as

x_1*x_2 = x_3;  
x_1*x_3 = x_4; 
......  
x_1*x_n = x_{n+1}

Many Thanks!

P.S. NON-commercial solver IPOPT can solve it efficiently, and I would like to try the best commercial solver and comparing them.

**Comment by johanlofberg** [comment, marked as answer]
there's no answer to such a question, you almost never know before testing

**Reply by 0xF0F0F0F** [reply]
Thank you Johan! Is there any recommended commercial solvers for non-convex problems? I would like to test them

**Reply by johanlofberg** [reply]
As I said, there are no recommended solvers as you never know until you've tried them. knitro is very fast on some, but so is fmincon on some, and snopt.
